src/mococrw/private/IOUtils.h

```diff
@@ -51,6 +51,13 @@ auto openSSLObjectFromFile(const std::string &filename)
     return Func({buffer.data(), buffer.size()});
 }
 
+template<class Res, Res(Func)(const std::string&, const std::string&)>
```

Comment:
Rather than requiring to specify the return type separately in Res, couldn't we use `std::result_of_t<>` from `type_traits` to automatically compute it? I'm thinking

```cpp
template<class F>
std::result_of_t<F&&(const std::string&, const std::string&)> openSSLObjectFromFile(F&& f, const std::string& filename, const std::string& password)
{
    auto buffer = bytesFromFile<char>(filename);
    return f({buffer.data(), buffer.size()}, password);
}
```

and then change the invocation to

```cpp
  return openSSLObjectFromFile(&mococrw::AsymmetricPrivateKey::readPrivateKeyFromPEM, filename, password);
```

Unfortunately `std::result_of_t<>` is undefined in C++11 when the given function cannot be invoked with the two `const std::string&` arguments, and only C++14 changes this to SFINAE, which is what we want in this case:

> As formulated in C++11, the behavior of std::result_of is undefined when INVOKE(std::declval<F>(), std::declval<ArgTypes>()...) is ill-formed (e.g. when F is not a callable type at all). C++14 changes that to a SFINAE (when F is not callable, std::result_of<F(ArgTypes...)> simply doesn't have the type member). 

Is MoCOCrW C++14 already? If not we'd probably have to protect this with a `std::enable_if_t<std::is_invocable_v<F(const std::string&, const std::string&)>> = 0`, except that `std::is_invocable` is C++17 only, so we'd have to either copy an implementation (https://stackoverflow.com/a/51188325) or use one from boost.